src/uri/uri_default_port.cr

```diff
@@ -0,0 +1,64 @@
+class URI
+  # Registry for URI scheme default ports.
+  private class DefaultPort
+    # A map of schemes and their respective default ports, seeded
+    # with some well-known schemes.
+    @default_port = {
+      "ftp"    => 21,
+      "ftps"   => 990,
+      "gopher" => 70,
+      "http"   => 80,
+      "https"  => 443,
+      "ldap"   => 389,
+      "ldaps"  => 636,
+      "nntp"   => 119,
+      "scp"    => 22,
+      "sftp"   => 22,
+      "ssh"    => 22,
+      "telnet" => 23,
+    }
+
+    # Returns the default port for the given `scheme` if known,
+    # otherwise returns `nil`.
+    #
+    # ```
+    # default_port = DefaultPort.new
+    # default_port["http"]  # => 80
+    # default_port["ponzi"] # => nil
+    # ```
+    def [](scheme : String?) : Int32?
+      return nil if scheme.nil?
+      @default_port[normalize(scheme)]?
```

Comment:
It feels weird to forward a `#[]` method to a `#[]?` method, I think it should be consistent.

Reply:
There should be a separate `#[]?(scheme : String)` method.